app/controllers/home_controller.rb

```diff
@@ -21,4 +21,38 @@ def post_comments
 			redirect_to root_url
     end
   end
+
+  def like_post_and_comment
+    @like = Like.find_by_likeable_id_and_likeable_type_and_user_id(params[:likeable_id], params[:likeable_type], current_user.id)
+    if !@like.present?
+      @like = current_user.likes.new(likeable_type: params[:likeable_type], likeable_id: params[:likeable_id])
+      respond_to do |format|
+        if @like.save
+          @message = 'success'
+          format.js
+          format.html { redirect_to root_url, notice: 'like was successfully created.'}
+        else
+          @message = 'invalid'
+          format.js
+          format.html { render 'new', notice: 'Something went wrong.' }
+        end
+      end
+    else
+      respond_to do |format|
+        if @like.destroy
+          @message = 'success'
+          format.js
+          format.html { redirect_to root_url, notice: 'like was successfully destroyed.'}
+        else
+          @message = 'invalid'
+          format.html { render 'new', notice: 'Something went wrong.' }
+        end
+      end
+    end
+  end
+
+  # private
+  # def like_params
```

Comment:
Why you're not using these in your actions? otherwise it insecure and anybody can utilize your action from outside world

Reply:
Using it but forgot to uncomment it.